app/src/main/java/com/skyyo/samples/features/textRecognition/cameraRecognition/CameraTextRecognitionScreen.kt

```diff
@@ -0,0 +1,109 @@
+import java.util.concurrent.Executors
+
+@OptIn(
+    ExperimentalPermissionsApi::class,
+    ExperimentalLifecycleComposeApi::class
+)
+@Composable
+fun CameraTextRecognitionScreen(viewModel: CameraTextRecognitionViewModel = hiltViewModel()) {
+    val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val recognizedSurfaceData by viewModel.recognizedSurfaceData.collectAsStateWithLifecycle()
+    val cameraPermissionState = rememberPermissionState(Manifest.permission.CAMERA)
+    val cameraExecutor = remember { Executors.newSingleThreadExecutor() }
+    val imageCapture = remember {
+        ImageCapture.Builder()
+            .setCaptureMode(ImageCapture.CAPTURE_MODE_MINIMIZE_LATENCY)
+            .build()
+    }
+    val imageAnalyzer by lazy {
+        ImageAnalysis.Builder()
+            .build()
+            .also {
+                it.setAnalyzer(
+                    cameraExecutor,
+                    TextImageAnalyzer(viewModel::onSurfaceAnalyzed)
+                )
+            }
+    }
+
+    DisposableEffect(Unit) {
+        onDispose { cameraExecutor.shutdown() }
+    }
+
+    PermissionRequired(
+        permissionState = cameraPermissionState,
+        permissionNotGrantedContent = {
+            Button(
+                onClick = { cameraPermissionState.launchPermissionRequest() },
+                modifier = Modifier.systemBarsPadding()
+            ) {
+                Text(text = "Add camera permission")
+            }
+        },
+        permissionNotAvailableContent = {
+            Button(onClick = { context.goAppPermissions() }, Modifier.systemBarsPadding()) {
+            Text(text = "Add camera permission from app Settings")
+        }
+        }
+    ) {
+        val cameraPreview = remember {
+            PreviewView(context).apply {
+                id = View.generateViewId()
+                scaleType = PreviewView.ScaleType.FIT_CENTER
+                implementationMode = PreviewView.ImplementationMode.COMPATIBLE
+            }
+        }
+        val cameraProvider by produceState<ProcessCameraProvider?>(null) {
+            value = tryOrNull { ProcessCameraProvider.getInstance(context).get() }
+        }
+
+        remember(cameraProvider) {
+            cameraProvider?.let { camProvider ->
+                camProvider.unbindAll()
+                camProvider.bindToLifecycle(
+                    lifecycleOwner,
+                    CameraSelector.DEFAULT_BACK_CAMERA,
+                    Preview.Builder()
+                        .build().also { preview ->
+                            preview.setSurfaceProvider(cameraPreview.surfaceProvider)
+                        },
+                    imageCapture,
+                    imageAnalyzer
+                )
+            }
+        }
+
+        DisposableEffect(Unit) {
+            onDispose { cameraProvider?.unbindAll() }
+        }
```

Comment:
Let's keep all effects as high as possible in appropriate scope - please move it to be under variables defined inside this `PermissionRequired.content` scope